lib/taskana-core/src/main/java/pro/taskana/TaskanaEngine.java

```diff
@@ -66,6 +67,26 @@
      */
     void setConnection(java.sql.Connection connection) throws SQLException;
 
+
+    /**
+     * Open the connection to the database. to be called at the begin of each Api call that accesses the database
+     */
+    void openConnection();
+
```

Comment:
openConnection, getSqlSession and returnConnection are calls that are not meant as public API. These calls are used by Taskana Implementation classes. Opening them for use by the customer may cause problems because usage has to adhere to a defined protocol.